django/test/client.py

```diff
@@ -675,7 +684,18 @@ def _handle_redirects(self, response, **extra):
             if not path.startswith('/'):
                 path = urljoin(response.request['PATH_INFO'], path)
 
-            response = self.get(path, QueryDict(url.query), follow=False, **extra)
+            if response.status_code in (HTTPStatus.TEMPORARY_REDIRECT,
+                                        HTTPStatus.PERMANENT_REDIRECT):
+                # 307 and 308 status codes need to preserve their request method
+                # across redirects.
+                request_method = response.request['REQUEST_METHOD'].lower()
+                req_method = getattr(self, request_method)
```

Comment:
It's a small detail, but I think the variable naming could be improved here. `request_method` and `req_method` are very similar names (one is just a shortening of the other) but they mean very different things. I might rename `request_method` to `req_method_name` or something. Or you could probably just turn this into a single line and avoid a temporary variable.

Reply:
Good point, I've made it a single line